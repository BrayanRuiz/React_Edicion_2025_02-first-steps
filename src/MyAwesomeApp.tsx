

export function MyAwesomeApp() {

    const firstName = 'Brayan';
    const lastName = 'Ruiz';

    const favoriteGames = ['Elden Ring', 'Smash', 'Metal Gear'];

    const isActive = true;

    const address = {
        zipCode: 53534,
        city: "Monterrey",
    }

    return (
        <div>
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>

            <p>{2 + 2}</p>
            <p className="mi-clase-favorita">{favoriteGames.join(', ')}</p>

            <h1>{isActive ? 'Activo' : 'Inactivo'}</h1>

            <p
                style={{ backgroundColor: 'red', }}
            >{JSON.stringify(address)}</p>
        </div>
    );
};