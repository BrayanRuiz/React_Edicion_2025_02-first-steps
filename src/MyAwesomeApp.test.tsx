import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from '@testing-library/react';

// ? 'screen' : Para poder trabajar el resultado de la renderizaciÃ³n en mi componente.

describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {
        render(<MyAwesomeApp />);
        screen.debug();

        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Brayan');

        // const h1 = container.querySelector('h1');
        // const h3 = container.querySelector('h3');
        
        // expect(h1?.innerHTML).toContain('Brayannn');
        // expect(h3?.innerHTML).toContain('Ruiz');
    });

    test('shourd march snapshot', () => {
        const { container } = render(<MyAwesomeApp />);

        expect(container).toMatchSnapshot();
    });

});