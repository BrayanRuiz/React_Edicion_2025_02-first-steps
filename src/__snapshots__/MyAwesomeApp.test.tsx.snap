// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > shourd march snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="first-name-title"
    >
      Brayan
    </h1>
    <h3>
      Ruiz
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Metal Gear
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: red;"
    >
      {"zipCode":53534,"city":"Monterrey"}
    </p>
  </div>
</div>
`;
